go-controller/pkg/ovn/base_network_controller_secondary.go

```diff
@@ -260,7 +260,8 @@ func (bsnc *BaseSecondaryNetworkController) ensureLocalZonePodForSecondaryNetwor
 func (bsnc *BaseSecondaryNetworkController) ensureRemoteZonePodForSecondaryNetwork(pod *kapi.Pod, addPort bool) error {
 
 	if !bsnc.doesNetworkRequireIPAM() {
-		return nil
+		_, err := bsnc.addPodToNamespaceForSecondaryNetwork(pod.Namespace, nil)
+		return err
```

Comment:
I see why we need to watch namespaces, but I don't understand why we need to react to the pod events with nil ips? the cached namespace will be created and updated based on namespace events only, and that is all we need?

Reply:
I had proposed introducing a dedicated method that doesn't care about IPs / address sets.

I had an offline call w/ @jcaamano where he asked me to re-use as much code as possible. 

Thus, I ended up re-using that method.

Reply:
Yeah I agree with the general approach, let me rephrase my question: what exactly will not work if we revert only this specific change in `ensureRemoteZonePodForSecondaryNetwork` and leave everything else as is?

Reply:
ah! My bad, didn't understand your original intention.

Seems you're right, and there's no need to actually keep track of these IPs for remote zones.

For the local zones, it **is** needed: we need to keep track of the LSP names for the port groups.

Good catch! 
@npinaeva